### ISSUE TITLE
Boolean expressions for httpQL queries

### ISSUE THREAD

**Issue by boulter** [HubSpot/httpQL#4]
Currently, all httpQL conditions are ANDed together. It would be super useful to expand this the rest of the boolean expression family - OR and NOT. 

We'll first need to figure out how to group and format these new expressions as a query string.

**Comment by nbelisle11** [comment]
I've noodled on this in the past and the biggest issue I have is there's no guaranteed order on query params. 

While there's no semantic support for including a request body in a GET there is some industry support for it ([ElasticSearch](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-request-body.html) for example)

This feels a little safer and also, certainly more human readable. Thoughts @boulter?

If we like this approach, I can try putting together a sample JSON spec that we can play with. The plumbing shouldn't be too difficult.

**Comment by boulter** [comment]
Yeah, it may not be possible to do using an unencoded query string. We could either encode the entire query as one parameter or allow some expressions to be more complex. For example:

`http://httql.org/?foo=bar&expr1=%28foo2%3D1%7Cfoo2%3D9%29%26bar2%3D7`

where expr1 decoded is `(foo2=1|foo2=9)&bar2=7`

**Comment by mjball** [comment]
re: get with body, https://git.hubteam.com/HubSpot/HubSpotConnect/issues/383

**Comment by nbelisle11** [comment]
Womp womp, thanks for the link @mjball

Encoding individual expressions then ANDing all expressions sounds pretty good @boulter
